File: Examples/CSharp/ProgrammersGuide/Load-Save-Convert/SaveVisioDiagram.cs
using Aspose.Diagram;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

namespace CSharp.ProgrammersGuide.Load_Save_Convert
{
    public class SaveVisioDiagram
    {
        public static void Run() 
        {
            //ExStart:SaveVisioDiagram
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_LoadSaveConvert();
            // load an existing Visio diagram
            Diagram diagram = new Diagram(dataDir + "Drawing1.vsdx");
            // save diagram using the direct path
            diagram.Save(dataDir + "SaveVisioDiagram_Out.vsdx", SaveFileFormat.VSDX);

            MemoryStream stream = new MemoryStream();
            // save diagram in the stream
            diagram.Save(stream, SaveFileFormat.VSDX);
            //ExEnd:SaveVisioDiagram
        }
    }
}

File: Examples/VisualBasic/Load-Save-Convert/SaveVisioDiagram.vb

Imports Aspose.Diagram
Imports System
Imports System.IO

Public Class SaveVisioDiagram
    Public Shared Sub Run()
        ' ExStart:SaveVisioDiagram
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_LoadSaveConvert()
        ' Load an existing Visio diagram
        Dim diagram As New Diagram(dataDir & Convert.ToString("Drawing1.vsdx"))
        ' Save diagram using the direct path
        diagram.Save(dataDir & Convert.ToString("SaveVisioDiagram_Out.vsdx"), SaveFileFormat.VSDX)

        Dim stream As New MemoryStream()
        ' Save diagram in the stream
        diagram.Save(stream, SaveFileFormat.VSDX)
        ' ExEnd:SaveVisioDiagram
    End Sub

End Class
